elpis/wrappers/objects/dataset.py

```diff
@@ -83,16 +94,85 @@ def load(cls, base_path: Path):
         return self
 
     @property
-    def tier(self) -> str:
-        return self.config['tier']
+    def files(self):
+        return self.config['files']
 
-    @tier.setter
-    def tier(self, value: str):
-        self.config['tier'] = value
+    @property
+    def tier_max_count(self) -> int:
+        return self.config['tier_max_count']
+
+    @tier_max_count.setter
+    def tier_max_count(self, value: int):
+        self.config['tier_max_count'] = value
 
     @property
-    def files(self):
-        return self.config['files']
+    def tier_order(self) -> int:
+        return self.config['tier_order']
+
+    @tier_order.setter
+    def tier_order(self, value: int):
+        self.config['tier_order'] = value
+
+    @property
+    def tier_types(self) -> List[str]:
+        return self.config['tier_types']
+
+    @tier_types.setter
+    def tier_types(self, value: List[str]):
+        self.config['tier_types'] = value
+
+    @property
+    def tier_type(self) -> str:
+        return self.config['tier_type']
+
+    @tier_type.setter
+    def tier_type(self, value: str):
+        self.config['tier_type'] = value
+
+    @property
+    def tier_names(self) -> List[str]:
+        return self.config['tier_names']
+
+    @tier_names.setter
+    def tier_names(self, value: List[str]):
+        self.config['tier_names'] = value
+
+    @property
+    def tier_name(self) -> str:
+        return self.config['tier_name']
+
+    @tier_name.setter
+    def tier_name(self, value: str):
+        self.config['tier_name'] = value
+
+
+    def get_elan_tier_attributes(self, input_dir: Path = Path('.')):
+        """
+        Iterate a dir of elan files and compiles info about all the files' tiers:
+        unique tier types, unique tier names, and the num of tiers
+        """
+        # Use sets internally for easy uniqueness, conver to lists when done
+        _tier_types: Set[str] = set()
+        _tier_names: Set[str] = set()
+        all_files_in_directory: Set[str] = set(
+            glob.glob(os.path.join(input_dir, "**"), recursive=True))
+        input_eafs_files: List[str] = [
+            file_ for file_ in all_files_in_directory if file_.endswith(".eaf")]
+        for file_ in input_eafs_files:
+            input_eaf = Eaf(file_)
+            for tier_type in list(input_eaf.get_linguistic_type_names()):
+                _tier_types.add(tier_type)
+                tier_ids: List[str] = input_eaf.get_tier_ids_for_linguistic_type(
+                    tier_type)
+                for tier_id in tier_ids:
+                    _tier_names.add(tier_id)
+            # count the number of tiers, use the max from all files
+            tier_count = len(list(input_eaf.get_tier_names()))
```

Comment:
I assume Elan doesn't allow duplicates in tier names?

Reply:
Yes it does. As discussed, let's show unique names in the GUI and when getting the transcriptions will use the first tier by index that matches the chosen name.

Reply:
It is case-sensitive, so `Phrase` and `phrase` are allowed, but not `Phrase` and `Phrase`.